src/btree/bt_delete.c

```diff
@@ -204,10 +204,24 @@ __wt_delete_page_rollback(WT_SESSION_IMPL *session, WT_REF *ref)
 			 */
 			__wt_free(session, ref->page_del->update_list);
 			__wt_free(session, ref->page_del);
-			goto done;
+			return;
 		}
+		/*
+		 * We wait for the change in page state, yield before
+		 * retrying, and if we've yielded enough times, start
+		 * sleeping so we don't burn CPU to no purpose.
+		 */
+		if (++yield_count < WT_THOUSAND) {
```

Comment:
yield_count increment can be moved inside the if condition block, so that when yield_count is above 1000, it doesnt unnecessarily increment.